src/client/cockpit-client

```diff
@@ -163,6 +184,45 @@ class ExternalCockpitWs:
         pass
 
 
+class InternalCockpitWs:
+    def start(self):
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.socket.bind(('127.0.90.90', 0))
+        self.socket.listen()
+
+        self.write_config()
+        self.ws = subprocess.Popen([f'{libexecdir}/cockpit-ws'],
+                                   preexec_fn=self.preexec_fn, pass_fds=[3])
+
+        host, port = self.socket.getsockname()
+        return f'http://{host}:{port}/'
+
+    def preexec_fn(self):
+        os.environ['XDG_CONFIG_DIRS'] = self.config_dir.name
+        os.environ['LISTEN_PID'] = str(os.getpid())
+        os.environ['LISTEN_FDS'] = str(1)
+        prctl(prctl.SET_PDEATHSIG, signal.SIGKILL)
+        os.dup2(self.socket.fileno(), 3)
+
+    def stop(self):
+        self.ws.kill()
+        self.config_dir.cleanup()
+
+    def write_config(self):
+        self.config_dir = tempfile.TemporaryDirectory(prefix='cockpit-client-', suffix='-etc')
+        os.mkdir(f'{self.config_dir.name}/cockpit')
+        with open(f'{self.config_dir.name}/cockpit/cockpit.conf', 'x') as cockpit_conf:
+            cockpit_conf.write(f'''# dynamically generated by cockpit-client
+                               [WebService]
```

Comment:
Looks a little odd to indent the config file so much (but should be harmless). There's [textwrap.dedent()](https://docs.python.org/3.6/library/textwrap.html#textwrap.dedent) for such cases.

(Feel free to entirely ignore)

Reply:
I had it left-aligned first, but the config parser seems happy enough...

I'll add dedent if you like